extensions/wikia/Search/classes/Hooks.php

```diff
@@ -3,161 +3,196 @@
  * Class definition for Wikia\Search\Hooks
  */
 namespace Wikia\Search;
+
 use Wikia\Search\Indexer;
+
 /**
  * This class is responsible for storing MediaWiki hook logic related to search.
  * Each method must be registered as a hook in the setup file, given the appropriate global settings.
  * We have also migrated backlink logic here.
+ *
  * @author relwell
  * @package Search
  */
-class Hooks
-{
+class Hooks {
 	/**
 	 * Stores backlinks for each target article, listing the backlink text, and which source articles link to them
-	 * @var unknown_type
+	 *
+	 * @var *
 	 */
-	protected static $outboundLinks = array();
-	
+	protected static $outboundLinks = [];
+
 	/**
 	 * Encapsulates MediaWiki logic
-	 * @var Wikia\Search\MediaWikiService
+	 *
+	 * @var \Wikia\Search\MediaWikiService
 	 */
 	protected static $service;
-	
+
 	/**
 	 * Sends delete request to article if it gets deleted
-	 * @param WikiPage $article
-	 * @param User $user
+	 *
+	 * @param \WikiPage $article
+	 * @param \User $user
 	 * @param integer $reason
 	 * @param integer $id
+	 *
+	 * @return true
 	 */
 	public static function onArticleDeleteComplete( &$article, \User &$user, $reason, $id ) {
-		return (new Indexer)->deleteArticle( $id );
+		return ( new Indexer )->deleteArticle( $id );
 	}
-	
+
 	/**
 	 * Reindexes the page
-	 * @param WikiPage $article
-	 * @param User $user
+	 *
+	 * @param \WikiPage $article
+	 * @param \User $user
 	 * @param string $text
 	 * @param string $summary
 	 * @param bool $minoredit
 	 * @param bool $watchthis
 	 * @param string $sectionanchor
 	 * @param array $flags
-	 * @param Revision $revision
+	 * @param \Revision $revision
 	 * @param int $status
 	 * @param int $baseRevId
+	 *
+	 * @return true
 	 */
-	public static function onArticleSaveComplete( &$article, &$user, $text, $summary,
-	        $minoredit, $watchthis, $sectionanchor, &$flags, $revision, &$status, $baseRevId ) {
-		return (new Indexer)->reindexBatch( array( $article->getTitle()->getArticleID() ) );
+	public static function onArticleSaveComplete(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameters "$user, $text, $summary, $minoredit, $watchthis, $sectionanchor, $flags, $revision, $status, $baseRevId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This function has 11 parameters, which is greater than the 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)